src/companion.rs

```diff
@@ -0,0 +1,136 @@
+use anyhow::Context;
+use regex::Regex;
+use tokio::process::Command;
+
+use crate::github_bot::GithubBot;
+
+pub async fn companion_update(
+	github_bot: &GithubBot,
+	owner: &str,
+	repo: &str,
+	branch: &str,
+) -> anyhow::Result<()> {
+	let token = github_bot.client.auth_key().await?;
+	Command::new("rustup")
+		.arg("update")
+		.spawn()
+		.context("spawn rustup update")?
+		.await
+		.context("rustup update")?;
+	Command::new("git")
+		.arg("clone")
+		.arg("-vb")
+		.arg(branch)
+		.arg(format!(
+			"https://x-access-token:{token}@github.com/{owner}/{repo}.git",
+			token = token,
+			owner = owner,
+			repo = repo,
+		))
+		.arg("repo")
+		.spawn()
+		.context("spawn git clone")?
```

Comment:
instead of cloning (which could be quite heavy) would it make sense to just fetch the recent changes?

Reply:
it will now only clone once